lib/src/main/proto/bio/terra/tanagra/underlay/underlay.proto

```diff
@@ -35,6 +36,9 @@ message Underlay {
   // The hierarchies for attributes with hierarchies in the underlay.
   repeated Hierarchy hierarchies = 9;
 
+  // The text search information for entities in the underlay.
+  repeated Text texts = 10;
```

Comment:
Is it correct to have this at the top level? Could it belong under `EntityFiltersSchema` instead?

If this level is correct, possibly it is worth renaming; from a quick scan it's not clear whether this pertains to:
1. the source dataset
2. the entity model
3. or, the mapping

I don't have great suggestions yet as I'm not sure I fully grok the scope

Reply:
I would say this pertains to the entity model, similar to how hierarchies pertain to the entity model. i.e. "this is the information you need to do text search/hierarchical queries".

I could have put it under `EntityFiltersSchema`, since that also holds a reference to the entity. But my understanding of the intent behind `EntityFiltersSchema` is that it would hold information for the UI about how to construct filters when making API requests. This is based on initial handoff conversations with Will and the example in the test `nautical` [underlay](https://github.com/DataBiosphere/tanagra/blob/08a8740352c8e142cd8b58a5293e8e595895cc01/api/src/test/resources/underlays/nautical.yaml#L408). So this information about where to find information required to construct text search queries doesn't really seem to fit there.

I'm certainly open to renaming to make the purpose of this clearer -- maybe `TextSearchInformation`?

Reply:
I can't come up with names that I love that much better than what you have here, especially pluralized. 

Some that don't quite fit (I don't think):
TextIndex
TextEngine

Fine with your suggestion of TextSearchInformation (I suppose that's already plural?) or keeping what you had.

Maybe just some restructuring of the underlay proto into related sections / groupings would address my concern. Down the road, also we could consider having a top-level SearchSupport section; but I'm not going to push for that as it would cause a mass indentation as currently structured (maybe more viable later if we support splitting underlay sections into multiple files)

Reply:
Agreed it would be good to do some restructuring of the underlay. In particular, breaking it into at least 2 separate groups of files would make sense to me: entity model, mapping of underlying dataset(s) to entity model. This might be good to do in preparation for driving the indexing workflows directly from the underlay definition. Regardless, I'll take your suggestion to do this as follow-on work instead of in this PR.

Reply:
Renamed Text -> TextSearchInformation.